### PR TITLE
Public Unit testing training

### PR DESCRIPTION
**Pull request by lasyasreepada**
Wrote a JUnit test for getCacheSize method in LruBitmapCache. Fixed a bug in the base class that was causing the unit test to fail.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
     <ConfirmationsSetting value="0" id="Add" />
     <ConfirmationsSetting value="0" id="Remove" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
+  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_7" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (stswidwinski) [inline .idea/misc.xml:40]: Let's not downgrade! :)

File: app/build.gradle [hunk 2]
```diff
@@ -27,4 +27,7 @@ dependencies {
     compile 'com.android.support:appcompat-v7:24.2.1'
     compile 'com.android.volley:volley:1.0.0'
     testCompile 'junit:junit:4.12'
+    testCompile 'org.mockito:mockito-core:1.10.19'
 }
+
```
Comment (stswidwinski) [inline app/build.gradle]: Trailing whitespace

**Review by stswidwinski — COMMENTED** [review]
Good job! Clean code and nothing much to complain about but for few minor whitespace issues!

File: .idea/misc.xml (hunk 1 above)
Comment (Parthib) [reply .idea/misc.xml:40]: Happened on my branch too

File: app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+package com.example.samplegallery.Utilities;
+
+import android.content.Context;
+import android.util.DisplayMetrics;
+
```
Comment (Parthib) [inline app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java]: I think this newLine is not necessary

File: app/src/main/java/com/example/samplegallery/APIKeys.java [hunk 4]
```diff
@@ -10,5 +10,5 @@
  * keys publically!
  */
 public class APIKeys {
-    public static final String FLICKR_API_KEY = "64ffdc1fd4b18211f4f98513848d2335";
+    public static final String FLICKR_API_KEY = "aaad3788ac68ea36800d8986f14c9dd3";
```
Comment (Parthib) [inline app/src/main/java/com/example/samplegallery/APIKeys.java]: API key is exposed!

File: app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+import org.junit.Test;
+import static org.junit.Assert.*;
```
Comment (Parthib) [inline app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java]: Static and nonstatic imports are mixed together

File: .idea/misc.xml (hunk 1 above)
Comment (lasyasreepada) [reply .idea/misc.xml:40]: Fixed the issue see similar comment on #7

File: app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java (hunk 3 above)
Comment (lasyasreepada) [reply app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java]: Deleted new line and organized my imports by static first and then alphabetically.

File: app/build.gradle (hunk 2 above)
Comment (lasyasreepada) [reply app/build.gradle]: Got rid of extra white space

File: app/src/main/java/com/example/samplegallery/APIKeys.java (hunk 4 above)
Comment (lasyasreepada) [reply app/src/main/java/com/example/samplegallery/APIKeys.java]: Changed it back to the fake API key. Have been reading about using git-update-index to have git ignore a file that has already been committed

File: app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java (hunk 5 above)
Comment (lasyasreepada) [reply app/src/test/java/com/example/samplegallery/Utilities/LruBitmapCacheTest.java]: Changed to be static followed by alphabetical non static imports

**Pull request closed without merging**